mita/openstack.py

```diff
@@ -157,84 +157,105 @@ def _get_flavor(self) -> NodeSize:
             logger.error(be)
             raise OpenStackDriverError("Encountered an unknown exception.")
 
-    def _has_free_ip_address(self, network: str) -> bool:
+    def _get_network_by_name(self, name: str) -> OpenStackNetwork:
         """
-        Return True if the given network has an IPv4 Address to lease.
+        Retrieve the OpenStackNetwork instance using the provided name.
 
-        Arguments:
-            network: String, The network UUID
+        Args:
+            name:   the name of the network.
 
         Returns:
-            bool - True on success else False
+            OpenStackNetwork instance referenced by the name.
 
-        Note: This method returns True only if the given network has more than 3
+        Raises:
+            ResourceNotFound: when the named network resource does not exist in the
+                              given OpenStack cloud
         """
-        try:
-            resp = self.driver_v2.network_connection.request(
-                "/v2.0/network-ip-availabilities/{}".format(network),
-            )
-        except BaseException as be:
-            logger.info(be)
-            return False
-
-        if resp.status != 200:
-            logger.debug("Failed to gather the network details.")
-            return False
+        url = f"{self.driver_v2._networks_url_prefix}?name={name}"
+        object_ = self.driver_v2.network_connection.request(url).object
+        networks = self.driver_v2._to_networks(object_)
 
-        _total = resp.object.get("network_ip_availability", {}).get("total_ips")
-        _used = resp.object.get("network_ip_availability", {}).get("used_ips")
+        if not networks:
+            raise ResourceNotFound(f"No network resource with name {name} found.")
 
-        return (_total - _used) > 3
+        return networks[0]
 
-    def _get_network(self, network: Optional[str] = None) -> OpenStackNetwork:
+    def _has_free_ip_address(self, net: OpenStackNetwork) -> bool:
         """
-        Return the provider network.
+        Return True if the given network has an ip address to lease.
+
+        This method determines if the provided network has more than 3 IP addresses to
+        lease. The buffer is in place to avoid failures during node creation. As in
```

Comment:
Thank you for explaining this. I never knew why this code worked like this.

Maybe we should rename this method with the plural form "addresses", like `_has_free_ip_addresses`.